### DISCUSSION TITLE
How to support Keybase cask with different downloads per architecture

### DISCUSSION THREAD

**Discussion by codezninja** [Homebrew/discussions#4126] [Writing Formulae/Casks] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 3.6.20-12-g4cf365f
ORIGIN: https://github.com/Homebrew/brew
HEAD: 4cf365f6c90aa2e00c6c8b6af8994037696db5ca
Last commit: 19 minutes ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 0656b851d3e70706fc1a548bea902567d74dce08
Core tap last commit: 35 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: vim
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 14.0.0 build 1400
Git: 2.39.1 => /opt/homebrew/bin/git
Curl: 7.85.0 => /usr/bin/curl
macOS: 13.1-arm64
CLT: 14.2.0.0.1.1668646533
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some taps are not on the default git origin branch and may not receive
updates. If this is a surprise to you, check out the default branch with:
  git -C $(brew --repo homebrew/cask) checkout master
```

### Description of issue

I'm trying to figure out how to support `Keybase` application for both x86 and arm64 architecture. From what I can tell Cask doesn't seem to have the same support for Formulae with bottles that support multiple architectures. So I was going down the road of creating a new cask `keybase-arm64`. But I then read the [cookbook](https://docs.brew.sh/Cask-Cookbook#finding-the-simplified-name-of-the-vendors-distribution) and saw that we didn't want to go that route of appending architecture to cask names.

So any support would be grateful on how to approach this problem.

## Download Source:

* https://prerelease.keybase.io

**Comment by gromgit** [comment, marked as answer]
See the `zoom` cask for an example: https://github.com/Homebrew/homebrew-cask/blob/master/Casks/zoom.rb

**Reply by codezninja** [reply]
thank you so much

**Comment by miccal** [comment]
https://github.com/Homebrew/homebrew-cask/pull/139841
